### PR TITLE
INTYGFV-13255: Added functionality for unregistering private practiti…

### PR DESCRIPTION
**Pull request by mikeyOne**
…oners with no certificates.

### CODE REVIEW

File: web/src/main/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerController.java [hunk 1]
```diff
@@ -18,66 +18,43 @@
...
+    public ResponseEntity<PrivatePractitionerDTO> getPrivatePractitioner(@PathVariable String personOrHsaId) {
+        return privatePractitionerService.getPrivatePractitioner(personOrHsaId);
     }
 
     @GetMapping("/file")
     public ResponseEntity<byte[]> getPrivatePractitionerFile() {
+        return privatePractitionerService.getPrivatePractitionerFile();
+    }
 
-        List<PrivatePractitioner> privatePractitionerList = ppIntegratedUnitsService.getAllPrivatePractitioners();
```
Comment (IamFriberg) [inline web/src/main/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerController.java:62]: Why was this removed?

File: web/src/main/java/se/inera/intyg/intygsadmin/web/service/PrivatePractitionerServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        this.ppIntegrationRestService = ppIntegrationRestService;
+        this.itIntegrationRestService = itIntegrationRestService;
+    }
+
+    private static final String YES = "Ja";
+    private static final String NO = "Nej";
+    private static final String COUNT_FAILURE_MESSAGE = "Uppgift om utfärdade intyg kunde inte hämtas. Prova igen om en stund.";
+
+    @Override
+    public ResponseEntity<PrivatePractitionerDTO> getPrivatePractitioner(String personOrHsaId) {
```
Comment (IamFriberg) [inline web/src/main/java/se/inera/intyg/intygsadmin/web/service/PrivatePractitionerServiceImpl.java:53]: Should we realy return HttpStatus from this service? This could return a privatePractitionerDTO :)

File: web/src/test/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerControllerIT.java [hunk 3]
```diff
@@ -66,4 +66,15 @@ public void testGetPrivatePractitionerFile() {
...
 
+    @Test
+    public void testUnregisterPrivatePractitionerOk() {
+        RestAssured.sessionId = getAuthSession(ADMIN_USER);
+
+        final String hsaId = "SE123456-X3";
+
+        given()
+            .when().delete(PRIVATE_PRACTITIONER_API_ENDPOINT + "/" + hsaId)
+            .then().statusCode(OK);
```
Comment (IamFriberg) [inline web/src/test/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerControllerIT.java:77]: Seems like we are missing a call to the method and verification :)

**Review by IamFriberg — CHANGES_REQUESTED** [review]
Some smaller comments :)

File: web/src/main/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerController.java (hunk 1 above)
Comment (mikeyOne) [reply web/src/main/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerController.java:62]: I didn't remove it. But since I thought that the controller had too much logic for a controller I moved this to the new service that I created, together with the updated getPrivatePractitioner-function. The behaviour has not changed.

File: web/src/test/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerControllerIT.java (hunk 3 above)
Comment (mikeyOne) [reply web/src/test/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerControllerIT.java:77]: Well it does make a call to the method and checks that the response is 200 (OK). Test fails if for example 404 (NOT_FOUND).

File: web/src/main/java/se/inera/intyg/intygsadmin/web/service/PrivatePractitionerServiceImpl.java (hunk 2 above)
Comment (mikeyOne) [reply web/src/main/java/se/inera/intyg/intygsadmin/web/service/PrivatePractitionerServiceImpl.java:53]: As I see it the alternative would be to have the if check in the controller instead and there determine which status to set. I choose to do like this but I am open to changing it...

File: web/src/main/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerController.java [hunk 4]
```diff
@@ -18,66 +18,43 @@
...
+        PrivatePractitionerService privatePractitionerService) {
+        this.ppIntegrationRestService = ppIntegrationRestService;
+        this.privatePractitionerService = privatePractitionerService;
     }
 
     @GetMapping("/{personOrHsaId}")
-    public ResponseEntity<PrivatePractitioner> getPrivatePractitioner(@PathVariable String personOrHsaId) {
-
-        PrivatePractitioner privatePractitioner = ppIntegratedUnitsService.getPrivatePractitioner(personOrHsaId);
-        if (privatePractitioner == null) {
```
[the remark this answers is not in the document]
Comment (mikeyOne) [reply web/src/main/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerController.java:52]: See below answer

File: web/src/main/java/se/inera/intyg/intygsadmin/web/service/PrivatePractitionerServiceImpl.java (hunk 2 above)
Comment (IamFriberg) [reply web/src/main/java/se/inera/intyg/intygsadmin/web/service/PrivatePractitionerServiceImpl.java:53]: Maybe throw an IllegalArgumentException?

**Review by IamFriberg — APPROVED** [review]
Snyggt!
